test-network-function/accesscontrol/suite.go

```diff
@@ -50,124 +48,94 @@ var _ = ginkgo.Describe(common.AccessControlTestKey, func() {
+			for _, testType := range allTests {
+				testFile, err := testcases.LoadConfiguredTestFile(common.ConfiguredTestFile)
+				gomega.Expect(testFile).ToNot(gomega.BeNil())
 				gomega.Expect(err).To(gomega.BeNil())
-				context := common.GetContext()
-				// Collect container facts
-				for _, factsTest := range podFacts.TestCase {
-					args := strings.Split(fmt.Sprintf(factsTest.Command, pod.Name, pod.Namespace), " ")
-					podTest := containerpkg.NewPod(args, pod.Name, pod.Namespace, factsTest.ExpectedStatus, factsTest.ResultType, factsTest.Action, common.DefaultTimeout)
-					test, err := tnf.NewTest(context.GetExpecter(), podTest, []reel.Handler{podTest}, context.GetErrorChannel())
-					gomega.Expect(err).To(gomega.BeNil())
-					gomega.Expect(test).ToNot(gomega.BeNil())
-					_, err = test.Run()
-					gomega.Expect(err).To(gomega.BeNil())
-					if factsTest.Name == string(testcases.ContainerCount) {
-						podFact.ContainerCount, _ = strconv.Atoi(podTest.Facts())
-					} else if factsTest.Name == string(testcases.ServiceAccountName) {
-						podFact.ServiceAccount = podTest.Facts()
-					} else if factsTest.Name == string(testcases.Name) {
-						podFact.Name = podTest.Facts()
-						gomega.Expect(podFact.Name).To(gomega.Equal(pod.Name))
-						if strings.Compare(podFact.Name, pod.Name) > 0 {
-							podFact.Exists = true
-						}
-					}
-				}
-				// loop through various cnfs test
-				if !podFact.Exists {
-					ginkgo.It(fmt.Sprintf("is running test pod exists : %s/%s for test command :  %s", podFact.Namespace, podFact.Name, "POD EXISTS"), func() {
-						gomega.Expect(podFact.Exists).To(gomega.BeTrue())
-					})
-					continue
-				}
-				for _, testType := range pod.Tests {
-					testFile, err := testcases.LoadConfiguredTestFile(common.ConfiguredTestFile)
-					gomega.Expect(testFile).ToNot(gomega.BeNil())
-					gomega.Expect(err).To(gomega.BeNil())
-					testConfigure := testcases.ContainsConfiguredTest(testFile.CnfTest, testType)
-					renderedTestCase, err := testConfigure.RenderTestCaseSpec(testcases.Cnf, testType)
-					gomega.Expect(err).To(gomega.BeNil())
-					gomega.Expect(renderedTestCase).ToNot(gomega.BeNil())
-					for _, testCase := range renderedTestCase.TestCase {
-						if !testCase.SkipTest {
-							if testCase.ExpectedType == testcases.Function {
-								for _, val := range testCase.ExpectedStatus {
-									testCase.ExpectedStatusFn(pod.Name, testcases.StatusFunctionType(val))
-								}
-							}
-							if testCase.Loop > 0 {
-								runTestsOnPod(podFact.ContainerCount, testCase, testType, podFact, context)
-							} else {
-								runTestsOnPod(testCase.Loop, testCase, testType, podFact, context)
-							}
-						}
+				testConfigure := testcases.ContainsConfiguredTest(testFile.CnfTest, testType)
+				renderedTestCase, err := testConfigure.RenderTestCaseSpec(testcases.Cnf, testType)
+				gomega.Expect(err).To(gomega.BeNil())
+				gomega.Expect(renderedTestCase).ToNot(gomega.BeNil())
```

Comment:
what if no test is specified ? the renderedTestCase will be an empty array ? or nil ?

Reply:
In every scenario this returns an array with all the tests. There is an internal SkipTest field that is marked for tests that are not present